app/rtc_compare.py

```diff
@@ -0,0 +1,126 @@
+'''
+Compare the two RTC products if they are equivalent
+Part of the codes are copied from PROTEUS SAS
+'''
+
+import argparse
+
+import h5py
+import numpy as np
+
+RTC_S1_PRODUCTS_ERROR_TOLERANCE = 1e-6
+
+def _get_parser():
+    parser = argparse.ArgumentParser(
+        description='Compare two DSWx-HLS products',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    # Inputs
+    parser.add_argument('input_file',
+                        type=str,
+                        nargs=2,
+                        help='Input data in HDF5')
+
+    return parser
+
+def compare_hdf5(hdf_obj_1, hdf_obj_2, key_in='/', compare_attr=True):
+    '''
+    Recursively traverse the Dataset and Attributes in the 1st HDF5 (the reference).
+    Try to find the same dataset in the 2nd HDF5 (the target).
+    Returns 0 if the reference and the target HDF5 are identical within threshold
+    Prints out the difference report when they are not.
+
+    Paremeters:
+    ----------
+    hdf_obj_1: Reference HDF5 object
+    hdf_obj_2: Target HDF5 object
+
+    Return:
+    -------
+    _ : 0 if the two HDF are identical; 1 otherwise
+
+    '''
+
+    if isinstance(hdf_obj_1[key_in], h5py.Group):
+        for key_1, _ in hdf_obj_1[key_in].items():
+            if key_1 in hdf_obj_2[key_in].keys():
+                compare_hdf5(hdf_obj_1, hdf_obj_2, f'{key_in}/{key_1}')
+            else:
+                print(f'Cannot fine {key_in}/{key_1} in the target object.')
+    else:
+        # Compare the dataset
+        # Detect the kind of data based on the shape
+        if len(hdf_obj_1[key_in].shape)==0:
+            print('Scalar value, shape:',hdf_obj_1[key_in].shape)
+            # TODO Use np.array_equal() to compare
+
+        elif len(hdf_obj_1[key_in].shape)==1:
+            print('1d vector, shape:',hdf_obj_1[key_in].shape)
+            # TODO Use np.array_equal() to compare
+
+        elif len(hdf_obj_1[key_in].shape)>=2:
+            # Sigle or multiband raster
+            print('Raster. shape:',hdf_obj_1[key_in].shape)
+            # TODO: Use compare_raster_dataset() to compare
```

Comment:
Maybe you can add an `else` and raise an exception? Not sure if the else would ever be reached

Reply:
I don't think it would be reached, but agree that it would be a good practice to add that. Instead of raising the exception, I made it to return value so that the comparison goes on to the end.